packages/orama/src/methods/search.ts

```diff
@@ -246,7 +248,12 @@ export async function search<AggValue = Result[]>(
   }
 
   if (typeof results !== 'undefined') {
-    searchResult.hits = results.filter(Boolean)
+    for (const result of results) {
+      if (!result) continue
+
+      result.id = getDocumentIdFromInternalId(orama.internalDocumentIDStore, +result.id)
+      searchResult.hits.push(result)
```

Comment:
This `push` increases the size of the result allocating again the vector. Is it really needed?

Reply:
Not really, I will just modify the `.id` and then set the `.hits = results`.